File: src-old/test/java/io/tripled/social/client/infrastructure/SystemDateTimeProviderTest.java
package io.tripled.social.client.infrastructure;

import org.junit.Before;
import org.junit.Test;

import static org.hamcrest.CoreMatchers.notNullValue;
import static org.junit.Assert.assertThat;

public class SystemDateTimeProviderTest {

  private SystemDateTimeProvider dateTimeProvider;

  @Before
  public void setUp() {
    dateTimeProvider = new SystemDateTimeProvider();
  }

  @Test
  public void returnsNow() {
    assertThat(dateTimeProvider.now(), notNullValue());
  }
}

File: src/test/kotlin/io/tripled/social/client/infrastructure/inmemory/SystemDateTimeProviderTest.kt
package io.tripled.social.client.infrastructure.inmemory

import io.tripled.social.client.infrastructure.time.SystemDateTimeProvider
import org.hamcrest.CoreMatchers
import org.hamcrest.MatcherAssert
import org.junit.jupiter.api.BeforeEach
import org.junit.jupiter.api.Test
import java.time.LocalDateTime

internal class SystemDateTimeProviderTest {
    private var dateTimeProvider: SystemDateTimeProvider? = null

    @BeforeEach
    fun setUp() {
        dateTimeProvider = SystemDateTimeProvider()
    }

    @Test
    fun returnsNow() {
        MatcherAssert.assertThat<LocalDateTime>(dateTimeProvider!!.now(), CoreMatchers.notNullValue())
    }
}